js/app/interfaces/card.js

```diff
@@ -384,30 +383,25 @@ var Card = new Lang.Interface({
     set_thumbnail_frame_from_model: function (frame) {
         frame.visible = this._set_media_class_from_model(frame);
 
-        if (!this.model.thumbnail_uri) {
-            let placeholder = new Gtk.Frame({ visible: true });
-            this._add_css_class(placeholder, 'no_thumbnail');
-            frame.add(placeholder);
-            return;
+        let placeholder = new Gtk.Frame({ visible: true });
+        let provider = new Gtk.CssProvider();
+
+        if (this.model.thumbnail_uri) {
+            provider.load_from_data(
+                `frame {
```

Comment:
This started to look familiar then I realized that I had done something similar before! https://phabricator.endlessm.com/T12679

At the time we were afraid that not loading the images from a stream might also have a big impact on performance. Is there a good way to measure that?

Reply:
We are currently using the sync version of new_from_stream() to create the pixbuf so this is actually the same since the CSS machinery also load images with new_from_stream()

See https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-22/gtk/gtkcssimageurl.c#L58

I guess it would not be that hard to use a pixbuf loader there to load images asynchronously but then we need to add a way for gtk to know when to redraw them.

Reply:
Maybe this is a good candidate for instrumenting with EosProfileProbe so that you can do a performance diff with @ebassi's branch with the probe diff tool? I believe your explanation, but this is just _the_ perfect use case for gathering some numbers.